core/services/workflows/monitoring.go

```diff
@@ -0,0 +1,85 @@
+package workflows
+
+import (
+	"context"
+	"fmt"
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder"
+	"github.com/smartcontractkit/chainlink-common/pkg/beholder/pb"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"google.golang.org/protobuf/proto"
+	"log"
+)
+
+var registerTriggerFailureCounter metric.Int64Counter
+
+func initMonitoringResources(_ context.Context) (err error) {
+	registerTriggerFailureCounter, err = beholder.GetMeter().Int64Counter("RegisterTriggerFailure")
+	if err != nil {
+		return fmt.Errorf("failed to register trigger failure: %s", err)
+	}
+	return nil
+}
+
+// TODO: tradeoff between having labels be variadic vs values be variadic?
+func sendLogAsCustomMessageWithLabels(ctx context.Context, labels map[string]string, format string, values ...interface{}) (err error) {
+	for labelKey, labelValue := range labels {
+		ctx, err = KeystoneContextWithLabel(ctx, labelKey, labelValue)
+		if err != nil {
+			return err
+		}
+	}
+
+	return sendLogAsCustomMessage(ctx, format, values...)
+
+}
+
+func sendLogAsCustomMessage(ctx context.Context, format string, values ...interface{}) error {
+	msg, err := composeLabeledMsg(ctx, format, values...)
```

Comment:
🤔 Why are we composing the labelled message here and then setting them below?

Are these the same labels or different ones?

IMO since we have a structured API, it would be fine (probably better actually) to emit the labels in a structured format rather than transforming them into a string.

So the KeystoneCustomMessage could be expanded to include a Labels map for example